### PR TITLE
sqoop 1.4.6

### PR DESCRIPTION
**Pull request by stuwil**

### CODE REVIEW

File: Library/Formula/sqoop.rb [hunk 1]
```diff
@@ -34,4 +34,10 @@ def caveats; <<-EOS.undent
     for Sqoop to work.
     EOS
   end
+
+  test do
+    ENV["JAVA_HOME"] = "$(/usr/libexec/java_home)"
```
Comment (bfontaine) [inline Library/Formula/sqoop.rb]: This won’t work, use `/usr/libexec/java_home`.chomp instead.

File: Library/Formula/sqoop.rb [hunk 2]
```diff
@@ -34,4 +34,9 @@ def caveats; <<-EOS.undent
     for Sqoop to work.
     EOS
   end
+
+  test do
+    ENV["JAVA_HOME"] = `/usr/libexec/java_home`.chomp
```
Comment (xu-cheng) [inline Library/Formula/sqoop.rb]: We can set JAVA_HOME for users in install:
`bin.env_script_all_files(libexec/"bin", Language::Java.java_home_env)`

**Comment by stuwil** [comment]
@bfontaine Would you like me to add your changes from #43262?

**Comment by bfontaine** [comment]
@stuwil Yes I think they’ll be necessary. Thanks!

File: Library/Formula/sqoop.rb [hunk 3]
```diff
@@ -22,7 +24,17 @@ def spoop_envs
   def install
     libexec.install %w[bin conf lib]
     libexec.install Dir["*.jar"]
+
+    # use coreutils' readlink instead of the OS X one to get the '-f' option
```
Comment (mikemcquaid) [inline Library/Formula/sqoop.rb]: Can you submit an issue upstream about this? Would be nice if it wasn't needed.

**Comment by mikemcquaid** [comment]
Test is failing (presumably it's trying to write to `~`). Any ideas? http://bot.brew.sh/job/Homebrew%20Pull%20Requests/32163/version=yosemite/testReport/junit/brew-test-bot/yosemite/test_sqoop/

File: Library/Formula/sqoop.rb [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
   desc "Transfer bulk data between Hadoop and structured datastores"
   homepage "https://sqoop.apache.org/"
-  url "https://www.apache.org/dyn/closer.cgi?path=sqoop/1.4.5/sqoop-1.4.5.bin__hadoop-2.0.4-alpha.tar.gz"
-  version "1.4.5"
-  sha256 "2f36ba52ae64f2f674780984aa4ed53d43565098f208a4fcbd800af664b1def9"
+  url "https://www.apache.org/dyn/closer.cgi?path=sqoop/1.4.6/sqoop-1.4.6.bin__hadoop-2.0.4-alpha.tar.gz"
+  version "1.4.6"
+  sha256 "d582e7968c24ff040365ec49764531cb76dfa22c38add5f57a16a57e70d5d496"
 
+  depends_on :java
```
Comment (DomT4) [inline Library/Formula/sqoop.rb]: Any specific version requirement here or?

**Comment by stuwil** [comment]
Working on a patch for upstream...

**Comment by stuwil** [comment]
Submitted a patch upstream. https://issues.apache.org/jira/browse/SQOOP-2531

File: Library/Formula/sqoop.rb (hunk 4 above)
Comment (stuwil) [inline Library/Formula/sqoop.rb]: 1.6-1.7 it seems, it gets executed via Hadoop. Note 1.8 isn't supported - any way to exclude or specify a range?

**Comment by mikemcquaid** [comment]
@stuwil Could you add a link to that in the formula and apply the patch? Thanks!

**Comment by stuwil** [comment]
@mikemcquaid Added, but note we're using Java binaries here - the patch I submitted upstream is for the build script that generates the affected scripts. I've had to apply the same change to a number of different files, but please let me know if there's a better approach.

Haven't looked into it too much, but their ant build process isn't playing nicely with OS X...

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**